backend/src/main/java/com/redhat/cloud/notifications/db/session/Invoker.java

```diff
@@ -0,0 +1,42 @@
+package com.redhat.cloud.notifications.db.session;
+
+import org.jboss.logging.Logger;
+
+import javax.persistence.PrePersist;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+
+class Invoker {
+
+    private static final Logger LOGGER = Logger.getLogger(Invoker.class);
+    private final Map<Class<?>, InvokerCache> cacheMap = new HashMap<>();
+
+    private static class InvokerCache {
+        private Method prePersist;
+    }
+
+    private <T> InvokerCache get(Class<T> aClass) {
+        return cacheMap.computeIfAbsent(aClass, _aClass -> {
+            InvokerCache cache = new InvokerCache();
+            for (Method method: aClass.getDeclaredMethods()) {
+                PrePersist prePersist = method.getAnnotation(PrePersist.class);
+                if (prePersist != null) {
+                    cache.prePersist = method;
+                }
+            }
+
+            return cache;
+        });
+    }
+
+    public <T> void prePersist(T instance) {
+        InvokerCache cache = get(instance.getClass());
```

Comment:
@gwenneg Added a cache for the `prePersist` (per class). Most of the introspection methods should only be called once. Leaving the `invoke` to be called.
I'm not really sure if this is a valid approach :D.